reposilite-plugins/javadoc-plugin/src/main/kotlin/com/reposilite/plugin/javadoc/DocExtractor.kt

```diff
@@ -0,0 +1,56 @@
+package com.reposilite.plugin.javadoc
+
+import com.reposilite.web.http.ErrorResponse
+import com.reposilite.web.http.errorResponse
+import io.javalin.http.HttpCode
+import panda.std.Result
+import panda.std.Unit
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.util.jar.JarFile
+
+internal class DocExtractor {
+
+    fun extractJavadoc(jarFilePath: Path, destination: Path): Result<Unit, ErrorResponse> {
+        val file = jarFilePath.toFile()
+        val dest = destination.toFile()
+
+        // Some checks, to make sure we're working with valid files/paths.
+        require(!file.isDirectory) { "JavaDoc jar file must not be a directory!" }
+        require(dest.isDirectory) { "Destination must be a directory!" }
+        check(file.name.contains("doc.jar")) { "Invalid javadoc.jar! Name must contain: \"doc.jar\"" }
```

Comment:
These methods may throw exception + there is no need to use `File` instead of Path, e.g.:
```kotlin
if (Files.isDirectory(jarFilePath) return errorResponse(BAD_REQUEST, "JavaDoc [...]")
```